OpenRA.Game/Primitives/SegmentStream.cs

```diff
@@ -172,5 +172,20 @@ public static Stream CreateWithoutOwningStream(Stream stream, long offset, int c
 			stream.Seek(offset, SeekOrigin.Begin);
 			return new MemoryStream(stream.ReadBytes(count));
 		}
+
+		public static ReadOnlySpan<byte> GetReadableData(Stream stream, long offset, int size)
```

Comment:
This method is a bit weird and I think should be moved to the one caller as a private helper.

- If you pass in a MemoryStream, we don't advance the stream position.
- You're calling this to try and avoid an alloc/copy, but if you don't have a MemoryStream we're going to alloc a new array. A caller might prefer a `TryGet...` method instead, in case it could reuse an existing buffer in the non-MemoryStream case.